### PR TITLE
Upgrade Hugo theme for documentation

### PR DESCRIPTION
**Pull request by sophokles73**
The Learn theme has been upgraded to latest minor version 2.5.0.
Also replaced deprecated Hugo function/property references.

### CODE REVIEW

**Comment by sophokles73** [comment]
@b-abel would you mind taking a look at this?

File: site/documentation/layouts/partials/header.html [hunk 1]
```diff
@@ -15,15 +15,19 @@
...
+    <link href="{{"css/atom-one-dark-reasonable.css" | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
     <link href="{{"css/theme.css" | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
     <link href="{{"css/hugo-theme.css" | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
     {{with .Site.Params.themeVariant}}
-      <link href="{{(printf "css/theme-%s.css" .) | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
+    <link href="{{(printf "css/theme-%s.css" .) | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
     {{end}}
+    {{ range .Site.Params.custom_css -}}
+    <link href="{{(printf "%s" .) | relURL}}{{ if $assetBusting }}?{{ now.Unix }}{{ end }}" rel="stylesheet">
+    {{- end }}
```
Comment (b-abel) [inline site/documentation/layouts/partials/header.html:26]: This looks like a useful new feature. We could leverage it by adding the following to the `[params]` section in `hono/site/documentation/config.toml` under :

```toml
  # Provide a list of custom css files to load relative from the `static/` folder in the site root.
  custom_css = ["css/hono.css", "css/theme-hono.css", "css/swagger-ui-3.23.11.css"]
```

**Review by b-abel — APPROVED** [review]

File: site/documentation/layouts/partials/header.html (hunk 1 above)
Comment (sophokles73) [reply site/documentation/layouts/partials/header.html:26]: To me it looks like `theme-hono.css` is already automatically loaded by the `header.html` partial and we only want to load the swagger stuff when we are on a page that actually contains a swagger UI. It is therefore being loaded by our `custom-header.html` in that case only. We could remove the loading of `hono.css` from `custom-header.html`, though ...

Comment (sophokles73) [reply site/documentation/layouts/partials/header.html:26]: @b-abel I have added the custom_css Site param ...

Comment (b-abel) [reply site/documentation/layouts/partials/header.html:26]: > To me it looks like `theme-hono.css` is already automatically loaded by the `header.html` partial and we only want to load the swagger stuff when we are on a page that actually contains a swagger UI. It is therefore being loaded by our `custom-header.html` in that case only. We could remove the loading of `hono.css` from `custom-header.html`, though ...

You are right. I had mainly the cache busting feature in mind and didn't look closely enough at the stylesheet loading aspect.

**Comment by b-abel** [comment]
LGTM

**Pull request merged**